image/git-init/git/git.go

```diff
@@ -336,3 +360,39 @@ func configSparseCheckout(logger *zap.SugaredLogger, spec FetchSpec) error {
 	}
 	return nil
 }
+
+type Operation[T any] func() (T, error)
+
+// RetryWithBackoff runs `operation` until it succeeds or the context is done,
+// with exponential backoff and jitter between retries.
+func RetryWithBackoff[T any](
```

Comment:
Any reason to export this ? it's only used in the package so we can keep it unexported.